### ISSUE TITLE
TCRMatch Blank output

### ISSUE THREAD

**Issue by trifle99** [IEDB/TCRMatch#56]
Hello
Im having some issues using TCRMatch. 
I tried using the TCRMatch Tool (http://tools.iedb.org/tcrmatch/) but I couldn't submit my file because I have about 20844 cdr3_b sequences that I would like to search. So I downloaded the TCRMatch and tried running it from my terminal.

My input file looks like this:
![image](https://github.com/user-attachments/assets/ab63e50e-6e06-4439-beb7-bb5a211cce8b)

I wanted to search for pairwise matches so this is the command I type into my terminal:
![image](https://github.com/user-attachments/assets/e6e136db-fbae-46be-877b-b34a279f34d3)

It shows me this:
![image](https://github.com/user-attachments/assets/89b029e3-fdf4-4bda-a961-85d389654e13)

And then when I check my output file, its empty:
![image](https://github.com/user-attachments/assets/3d67ceff-d834-434f-a929-4f88beafd7f7)

When I search for it against the IEDB database by running this:
![image](https://github.com/user-attachments/assets/4f75ba21-e729-46c2-a219-3dbfe028bed0)

My terminal looks like this:
![image](https://github.com/user-attachments/assets/2344cc35-c622-4683-867c-1f1a0d40c540)

But when I check my output file this time, it has columns but the rest is empty:
![image](https://github.com/user-attachments/assets/8e0e72e9-7806-4386-bf66-159f744c774e)

I made another text file with just the first 10 cdr3b sequences and I tried doing a pairwise match with that (nothing popped up in my terminal either) and I managed to get results:
![image](https://github.com/user-attachments/assets/8617fcd9-f195-4661-b380-f5dddf49650b)

Is the dataset I'm inputting too large? If so am I just going to have to take small samples and input them one by one?
I tried making an account at VDJServer as well but I am still waiting for a verification email to verify my account.
Please could I have some guidance, thanks!

**Comment by raphaeltrevizani** [comment]
This is the error message you get if your sequence contains invalid input characters, so while your input file _looks_ right, it's possible that it contains some end line characters or anything invalid. Can you attempt to replicate this error with a minimum amount of input peptides and then send the problematic input file here?

**Comment by trifle99** [comment]
It seems my input file is the problem as when I do lines from my input file using 'head -n input_file > output_file', I get the same error of 'at position x. Skipping.n (sequence)' and the output_file is empty apart from the column headers. 
However when I type out my sequences manually into a text file and run ./tcrmatch on it, I receive no errors and the output file contains results. 

This is the problematic input file in question:
[cdr3b_df.csv](https://github.com/user-attachments/files/16530105/cdr3b_df.csv)

I originally obtained this file by running [scirpy.get.airr_context](https://scirpy.scverse.org/en/latest/generated/scirpy.get.airr_context.html) on my annotated dataset and separating the cdr3b sequence into a pandas dataframe, I then trimmed the leading C and ending F/W from the sequences and exported that dataframe into my input csv file.

Im confused as to why my sequences won't work as I've checked the ending positions for any hidden spaces or characters but I don't see anything, but when I type them out manually into a textfile it seems to work?

**Comment by raphaeltrevizani** [comment]
Thank you for including the input file. I've successfully replicated your error. I checked your input file and 

1. There are several lowercase _a_ at the end of the file. I imagine these are artifacts, so I removed them.

2. Running
`cat -v cdr3b_df.csv`
shows many `^M` at the end of each line, which is a carriage return character (maybe this file was created in Windows and you're trying to run TCRMatch on Linux?). 

3. I removed this using a tool called dos2unix:
`dos2unix cdr3b_df.csv`
which generated a new file ([attached](https://github.com/user-attachments/files/16532453/unicode_cdr3b_df.csv)). I successfully ran a sample on TCRMatch ([also attached](https://github.com/user-attachments/files/16532466/unicode_cdr3b_df_10lines.csv)), so let me know if you can run the whole thing.

**Comment by trifle99** [comment]
Thanks, that helped a lot and I'm now getting results in my files!

(1. The a's were from sequences which I recieved 'nan' values for and when I trimmed the leading C's and ending F/W's it also trimmed the n's because I was removing the first and last characters of the sequence string lol

2. Yea, I created my cdr3b_df file in windows but to use TCRMatch I used mobaxterm to ssh into a linux HPC and used scp to transfer the cdr3b df file from my local windows pc to the remote linux HPC. I didn't know this would change the file!

3. I just tried dos2unix on my own file in the HPC and it fixed the errors I was getting)

Thank you very much for your help, it was very quick and helpful!
Have a great week! :]
